src/main/java/org/jabref/gui/dbproperties/DatabasePropertiesDialog.java

```diff
@@ -167,13 +167,42 @@ public void actionPerformed(ActionEvent e) {
         am.put("close", closeAction);
 
         ok.addActionListener(e -> {
-            storeSettings();
+            if (propertiesChanged()) {
+                storeSettings();
+            }
             dispose();
         });
 
         cancel.addActionListener(e -> dispose());
     }
 
+    private boolean propertiesChanged() {
+        Charset oldEncoding = panel.getBibDatabaseContext().getMetaData().getEncoding()
+                .orElse(Globals.prefs.getDefaultEncoding());
+        Charset newEncoding = (Charset) encoding.getSelectedItem();
+        boolean saveActionsChanged = fieldFormatterCleanupsPanel.hasChanged();
+        boolean saveOrderConfigChanged = newAndOldOrderConfigIsSame(getNewSaveOrderConfig(), oldSaveOrderConfig);
+        boolean changed = saveOrderConfigChanged || !newEncoding.equals(oldEncoding)
+                || !oldFileVal.equals(fileDir.getText()) || !oldFileIndvVal.equals(fileDirIndv.getText())
+                || (oldProtectVal != protect.isSelected()) || saveActionsChanged;
+        return changed;
+    }
+
+    private boolean newAndOldOrderConfigIsSame(SaveOrderConfig newSaveOrderConfig, SaveOrderConfig oldSaveOrderConfig) {
+        return newSaveOrderConfig.equals(oldSaveOrderConfig) ? false : true;
```

Comment:
This is equivalent to `!newSaveOrderConfig.equals(oldSaveOrderConfig)`, right? In this case you can also directly inline this code.